docs/source/nutshell.rst

```diff
@@ -0,0 +1,223 @@
+than the incoming one, the editor will signal the error.
+
+.. _label_t_signature_modeling:
+
+"""""""""""""""""""""""""""""""
+Signature verification 
+"""""""""""""""""""""""""""""""
+
+The output script of ``T`` is naïve, since anyone can produce the right witness.
+Usually, one wants to transfer bitcoins to a specific user.
+For instance, the following piece of code makes the 50 BTC in ``T1``
+redeemable only by user Alice: 
+
+.. code-block:: btm
+
+	const addrA = wif:mvMKu24MRsSDsu6LatViZtx64J6Cxsz3qe // Alice's address
+
+	transaction T1 {
+		input = T: 42
+		output = 50 BTC: fun(x) . versig(addrA; x)
+	}
+
+
+The constant ``addrA`` declares Alice's *address*:
+basically, it is the hash of Alice's public key.
+The address is prefixed by the keyword ``wif:``,
+which means that it is represented in the
+:doc:`Wallet Import Format types <types>` [#f1]_.
+Note that users may generate as many addresses as they want.
+
+The :ref:`predicate <label_c_functions>` ``versig(addrA; x)``
+in the output field of ``T1`` 
+evaluates to true if ``x`` is a valid signature
+of the transaction which redeems ``T1``,
+done with Alice's private key. 
+
+The transaction ``T1`` can be redeemed by a transaction ``T2`` made as follows:
+
+.. code-block:: btm
+
+	const skA = wif:cMpBA2wEGKrN1vpxSBzWWzFBwnrypR4rBev4vTbEDPHMeSMyyyMi // Alice's private key
+
+	transaction T2 {
+		input = T1: sig(skA)
+		output = 50 BTC: ... // any condition chosen by Alice
+	}
+
+The witness ``sig(skA)`` is the :ref:`signature <label_c_functions>`
+of transaction ``T2`` (without considering the witness itself)
+using the private key ``skA``.
+
+To perform examples with Bitcoin addresses and keys, one can use the online key generator in
+the sidebar of the  editor page.
+
+.. figure:: _static/img/sidebar.png
+	:scale: 100 %
+	:class: img-border
+	:align: center  
+
+.. _label_t1_modeling:
+
+"""""""""""""""""""""""""""""""
+Multiple inputs and outputs
+"""""""""""""""""""""""""""""""
+Transactions can have more than one output, in order to split the money on different recipients. 
+For instance, we might modify transaction ``T2`` to send some of the total bitcoins to  another address,  still under our control,  like this:
+
+.. code-block:: btm
+
+	const myAddress = wif:mvMKu24MRsSDsu6LatViZtx64J6Cxsz3qe
+        const mySecondAddress= wif:n3A4KGgZD9bW6k2pPccN4rUfX3CgYCPERb
+
+	transaction T1 {
+		input = T: 42
+		output = [
+                         40 BTC: fun(x) . versig(myAddress; x);
+                         10 BTC: fun(x) . versig(mySecondAddress; x)
+			]
+	}	
+
+
+In this transaction, the output field has two items, which can be redeemed separately. 
+
+Transactions can have more that one input, in case they need to gather money from several sources.
+For each input, the transaction must provide a suitable witness. In case inputs refers to transaction with multiple outputs, their outputs are numbered starting from 0. 
```

Comment:
refers to transaction -> refers to a transaction

Reply:
ok